fix(pluto.jl): update base to 2023-12-25

bryanpaget opened the pull request:
I updated the base system to 2023-12-25.

NOTE: Images are hosted on https://quay.io/repository/jupyter/datascience-notebook from now on.

/closes #531

bryanpaget commented:
Tested on Dev but will need testing on Prod due to nature of bug only affecting Prod.

bryanpaget commented:
Pluto.jl works on prod now:

![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/8212170/ecceab9e-b1c2-4625-9d65-47dafe59a284)

bryanpaget commented:
I also tested the RStudio image and it seems to be fine. Some packages were updated so I felt the need to check on it.

mathis-marcotte reviewed (changes requested):
Seems like jupyterlab-plotly and jupyterlab-dash aren't being installed correctly or something(maybe the version is off), they are getting the red X instead of the green OK. 
![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/84033116/fbb622ec-20f2-4de9-b3b5-1c69ee643549)

This is from the v1 image
![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/84033116/0eef2363-c57b-4525-b537-8b96c90e4c35)

bryanpaget commented:
Sweet news: jupyter-dash is no longer required as it is now built into JupyterLab.

![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/8212170/fb23484f-e1db-4da8-8386-9d6507327b56)

bryanpaget commented:
When I removed Jupyter Dash and replaced it with Dash, a funny thing happened. Dash version 0.39.0 is installed by default, so I'm going back to pin version 2.14.2 which is the latest version of Dash. Not sure why anything would cause 0.39.0 (released in 2019) to be installed...

bryanpaget commented:
Outcomes from this ticket:

- Pluto.jl was fixed and works on prod now.
- I discovered we had an old version of Plotly Dash installed from the plotly anaconda repo which is broken. I replaced it with the latest version which works better but seems to have some kind of networking issue making it not work in JupyterLab but at least it's the latest version so people can actually develop with it. The version we had from the plotly repo was version `0.39.0` which doesn't even have the html widgets so declarations like `dash.html.Div()` would not work! We now have version `2.14.2`, so this is a big improvement.
- JupyterLab was upgrade to version 4 and I discovered Plotly itself works just fine with version 4 of JupyterLab.

**TODO:** fix Dash! It was never working and now we are an inch closer! Versions `2.11` and newer of Dash are supposed to work with JupyterLab 4+.

mathis-marcotte reviewed (approved):
lgtm
